src/app/core/utils/geo.js

```diff
@@ -840,7 +840,9 @@ const geoutils = {
           const zip = new JSZip();
           const buffer = await data.arrayBuffer(data);
           zip.load(buffer);
-          const kmlFile = zip.file(/.kml$/i)[0];
+          const kmlFiles = zip.file(/.kml$/i);
+          // get last kml file (when doc.kml file has a reference to kml inside another folder)
+          const kmlFile = kmlFiles[kmlFiles.length -1];
```

Comment:
## TODO

Check if solution works with multiple network links, eg:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<kml xmlns="http://www.opengis.net/kml/2.2">
  <Document id="root_doc">
    <NetworkLink>
      <Link>
        <href>layers/file_2.kml</href>
      </Link>
    </NetworkLink>
    <NetworkLink>
      <Link>
        <href>layers/file_2.kml</href>
      </Link>
    </NetworkLink>
  </Document>
</kml>
```

Reply:
@Raruto i think it doesn't work for multiple kml files. I think we can close it and open another feature issue to improve this type of situation.For multiple kml files in kmz, we have to modify createVectorLayerFromFile method of src/core/utils/geo.js
because at moment return just one layer